naima/models.py

```diff
@@ -431,3 +431,87 @@ def __call__(self, e):
         e = _validate_ene(e)
         interpy = np.power(10, self._interplogy(np.log10(e.to('eV').value)))
         return self.amplitude * interpy * self.unit
+      
+      
+class AbsorptionModel(object):
+    """
+    A Table-model like class containing the different absorption models.
+
+    It returns dimensionless opacity values, that could be multiplied to any model.
+    
+    Parameters
+    ----------    
+    redshift : float
+        Redshift considered for the absorption evaluation.
+        The minimum value used is z = 0.01; Lower values will
+        be considered as redshift 0.
+        
+    ebl_absorption_model : string 
+        Name of the EBL absorption model to use. It may be:
+    
+    * A string equal to ``Dominguez`` (default), containing Dominguez 2011 EBL model
+    
+    NOTE: Current implementation does NOT perform an interpolation in the redshift, so 
+    it just uses the closes z value from the finely binned tau_dominguez11.out file 
+    (delta_z=0.01)    
+        
+    """
+
+    def __init__(self, redshift, ebl_absorption_model='Dominguez'):
+        from scipy.interpolate import interp1d
+        from astropy.io import ascii
+                    
+        if ebl_absorption_model == 'Dominguez':
+            converters = dict()
+            for i in range(0,500):
+               converters["col%s" % i] = [ascii.convert_numpy(np.float64)]
+            taus_table = ascii.read('naima/data/tau_dominguez11.out', converters=converters)
```

Comment:
If you are only reading the matrix from the file, using `np.loadtxt` might be easier than using io.ascii and building 500 converters. This is also not generalizable to other files if you want to add other models in the future.